### ISSUE TITLE
Regarding the issue of the environment variable  NUXT_HUB_USER_TOKEN

### ISSUE THREAD

**Issue by imba97** [nuxt-hub/cli#28]
I use Github Actions

I have tried many times and kept getting this warning：`Please login to deploy your project or set the NUXT_HUB_USER_TOKEN environment variable.`

I have already set up `NUXT_HUB_PROJECT_KEY` and `NUXT_HUB_USER_TOKEN`

```
Run npx nuxthub@0.5.18 deploy --no-build
  npx nuxthub@0.5.18 deploy --no-build
  shell: /usr/bin/bash -e {0}
  env:
    PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
    STORE_PATH: /home/runner/setup-pnpm/node_modules/.bin/store/v3
    NUXT_HUB_PROJECT_KEY: ***
    NUXT_HUB_USER_TOKEN: ***
  
npm warn exec The following package was not found and will be installed: nuxthub@0.5.18
[log] NuxtHub CLI
[info] Please login to deploy your project or set the `NUXT_HUB_USER_TOKEN` environment variable.
Error:  nuxthub login is not supported in headless mode yet.
```

---

I have set up Repository secrets and can confirm that they are correct

![image](https://github.com/user-attachments/assets/cfe8a576-8b8d-42db-a145-810d531a2a3d)

I wrote a simple script to print these values

```ts
import process from 'node:process'

console.log('NUXT_HUB_PROJECT_KEY', process.env.NUXT_HUB_PROJECT_KEY?.length)
console.log('NUXT_HUB_USER_TOKEN', process.env.NUXT_HUB_USER_TOKEN?.length)

console.log('NUXT_HUB_USER_TOKEN', process.env.NUXT_HUB_USER_TOKEN?.slice(-10))
```

but

![image](https://github.com/user-attachments/assets/4ba2ba5b-af2a-4d96-92fb-f49095a8e515)

**Comment by RihanArfan** [comment]
Could you check whether the token is 100% correct? That error is shown if fetching details fails too, not just if it's unset. https://github.com/nuxt-hub/cli/blob/529bd86771018cc13df35217ed5eddd978d9e203/src/commands/deploy.mjs#L47-L48

**Comment by imba97** [comment]
Yes, the token is correct.

I tried the following and the problem was solved, but it was weird

I modified data.mjs and printed `process.env` before `npx nuxthub deploy --no-build`

> copy-file.ts

```ts
import fs from 'fs-extra'

fs.copyFileSync(
  'scripts/data.mjs',
  'node_modules/nuxthub/src/utils/data.mjs'
)
```

> data.mjs

```diff
export const $api = ofetch.create({
  baseURL: joinURL(NUXT_HUB_URL, '/api'),
  onRequest({ options }) {
    options.headers = options.headers || {}
+   console.log('token')
+   console.log(`Bearer ${loadUserConfig().hub?.userToken || process.env.NUXT_HUB_USER_TOKEN || ''}`)
    if (!options.headers.Authorization) {
      options.headers.Authorization = `Bearer ${loadUserConfig().hub?.userToken || process.env.NUXT_HUB_USER_TOKEN || ''}`
    }
  },
  onResponseError(ctx) {
    if (ctx.response._data?.message) {
      ctx.error = new Error(`- ${ctx.response._data.message}`)
    }
  }
})

export function fetchUser() {
+ console.log('NUXT_HUB_USER_TOKEN', process.env.NUXT_HUB_USER_TOKEN)
+ console.log('env', process.env)
  if (!loadUserConfig().hub?.userToken && !process.env.NUXT_HUB_USER_TOKEN) {
    return null
  }
  return $api('/user').catch(() => {
    return null
  })
}
```

I was confused but it worked in the end

![image](https://github.com/user-attachments/assets/0ca657eb-b061-4573-817a-a432302c69ba)

![image](https://github.com/user-attachments/assets/1b0399d5-8fdf-4384-9d15-2f9df89e7681)

---

The above is yesterday's attempt. Today I restored the above code and it was successful.

![image](https://github.com/user-attachments/assets/cfc81c35-3ba7-4b14-9a51-3d1f9504e3f0)

But I never changed the token

![image](https://github.com/user-attachments/assets/1c3a5dc5-06c6-4498-87e5-8107ddad9b3c)

**Comment by atinux** [comment]
So finally it is working without any changes?

I will have a look at what could have make this issue.

**Comment by atinux** [comment]
Could you try the [v0.6.2](https://github.com/nuxt-hub/cli/releases/tag/0.6.2) with:

```bash
DEBUG=1 nuxthub deploy
```

**Comment by atinux** [comment]
It is now fixed in `v0.6.3`, sorry about this, it was related to an upgrade of ofetch
